bridge/engine/Engine.h

```diff
@@ -16,7 +17,7 @@ namespace bridge
 class EngineMixer;
 class EngineMessageListener;
 
-class Engine
+class Engine : private EngineThreadContext
```

Comment:
I didn't use the composition to not expose `processEngineTasks`. By using a private inheritance we can have `processEngineTasks` `protected` and it will not be visible that `Engine` is `EngineThreadContext ` for outside.

Use composition I guess we need to split `EngineThreadContext` in two class with inheritance also. On top we will have only the post methods. And on the button the methods that will process the queue so the type you give to outside only have access to `post` methods